Content.Server/Mind/MindSystem.cs

```diff
@@ -94,6 +94,11 @@ private void OnShutdown(EntityUid uid, MindComponent mind, ComponentShutdown arg
                     {
                         spawnPosition = _gameTicker.GetObserverSpawnPoint();
                     }
+                    else if (spawnPosition.Position == Vector2.Zero)
+                    {
+                        Transform(spawnPosition.EntityId).AttachToGridOrMap();
+                        spawnPosition = Transform(spawnPosition.EntityId).Coordinates;
+                    }
```

Comment:
True. I was mistakenly focused on doing the logic here where the entity is already deleted, instead of above where the original entity can still invoke the attach method itself.